---
title: "Summary"
author: "Tilen Gimpelj, Maja Kovač"
date: "`r date()`"
output: 
  html_document: 
    toc: yes
    toc_float: TRUE
    code_folding: hide
  bibliography: reference.bib
csl: ieee-with-url.csl
link-citations: yes
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,include=T,error=F,warning = F,message=F)
pkgs <- c("rlang","tidyverse","readxl","writexl","lubridate","stringr","esquisse","miniUI","pheatmap","RColorBrewer","hexbin","BiocManager","reshape2","ggnewscale","ggbeeswarm","kableExtra", "DT")
bioc_pkgs <- c("DESeq2","SummarizedExperiment","GEOquery","biomaRt","org.Hs.eg.db","org.Mm.eg.db","pathview","clusterProfiler","EnhancedVolcano","apeglm","genefilter","tximport","tximeta","GenomicFeatures","edgeR")

for (pkg in pkgs) {if (pkg %in% installed.packages()) {message(pkg," installed, loading...");library(pkg,character.only = T)} else {message(pkg," not instlled, installing...");install.packages(pkg,dependencies =T,quiet = T)}}
bmngr<-"BiocManager";
if (bmngr %in% installed.packages()) {library(bmngr,character.only = T)} else {install.packages(bmngr,dependencies =T,quiet = T)}
for (bpkg in bioc_pkgs) {if (bpkg %in% installed.packages()) {message(bpkg," installed, loading...");library(bpkg,character.only = T)} else {message(bpkg," not instlled, installing...");BiocManager::install(bpkg,update=T,ask=F)}}
```
<p><a class="btn btn-primary" data-toggle="collapse" href="#collapseExample1" role="button" aria-expanded="false" aria-controls="collapseExample1">
Environment
</a></p><div class="collapse" id="collapseExample1"><div class="card card-body">
```{r, results='asis',echo=FALSE}
info <- sessionInfo()
datatable(data.frame(version = sapply(info$otherPkgs,FUN = \(x) {c(x$package,x$Version)}),
                        "Package_Manager" = sapply(info$otherPkgs,FUN = \(x){if (x$Package %in% bioc_pkgs){"BiocManager"} else {""}})),
          caption = info$R.version$version.string)
```
</div></div>

# Summary

TODO
- tuki je treba napisat kratek opis članka in kaj študija iskala 

## Wenge Li and colleagues, the findings include:

- Diet choice can significantly affect stem cell function, including proliferation, differentiation, and senescence.

- Different types of diets, such as high-fat diets, low-calorie diets, and fasting, have different effects on stem cells.

- The effects of diet on stem cells are mediated by various mechanisms, such as oxidative stress, inflammation, and epigenetic changes.

- The choice of animal models and cell types used in stem cell research can also affect the interpretation of the results and the generalizability of the findings.

- The quality and clarity of reporting in scientific publications are crucial for facilitating reproducibility and transparency in stem cell research, especially regarding the details of diet selection and administration.

***

```{r}
GEOaccN <- "GSE151498"
```

Geo Access number: ``r GEOaccN``

# Preparation of data

```{r}
gse1 <- getGEO(GEOaccN,GSEMatrix =TRUE)[[1]]
se1 <- as(gse1, "SummarizedExperiment")
if (!(GEOaccN %in% list.files())){ getGEOSuppFiles(GEOaccN)}

files <- list.files(GEOaccN,full.names=T)
read.table(files[1], skip=0, sep = "\t", header=TRUE) -> df1 #, row.names = 1)
read.table(files[2], skip=0, sep = "\t", header=TRUE) %>%  
  rename("ensemblID"= "geneNames") %>% 
  full_join(df1 %>% 
       select(geneNames,ensemblID)) %>% 
  relocate(geneNames)-> df2#, row.names = 2)

```

## Dataset 1

```{r}

datatable(df1)
```


## Dataset 2

```{r}

datatable(df2)
```



## Genes  in the dataset:
<p><a class="btn btn-primary" data-toggle="collapse" href="#collapseExample2" role="button" aria-expanded="false" aria-controls="collapseExample2">
Genes
</a></p><div class="collapse" id="collapseExample2"><div class="card card-body">
```{r}
df1 %>%  
  group_by(geneNames) %>% 
  dplyr::count(geneNames)  %>% 
  rename(count = n, "Gene Name" = geneNames) %>% 
  datatable()  

```
</div></div>
***

## Transcription

mutation of dataset from given to genes as rownames and samples as columns

```{r}
df1 %>% 
  group_by(geneNames) %>% 
  summarise_if(is.numeric, sum) %>% 
  as.data.frame -> df1a

df1a %>% 
  column_to_rownames(var = "geneNames") -> df1b
```

```{r}
rownames(se1@colData)<-colnames(df1b)
df1b %>% 
  mutate_if(is.numeric,round) -> df1b

se1@colData$characteristics_ch1.3 <- factor(
  str_replace(
    str_replace(
      str_replace(
        str_extract(se1@colData$characteristics_ch1.3,
                    ": [A-Za-z 0-9]*"),
        ": ",
        ""),
      " $",
      ""),
    " ",
    "_"))

se1@colData$characteristics_ch1.5 <- 
  factor(
    str_replace(
      str_extract(
        se1@colData$characteristics_ch1.5,
        " [a-z]*"),
      " ",
      ""))
```

### Saving as DESeqDataSet
```{r}
dds_diet <- DESeqDataSetFromMatrix(countData = df1b,
                                     colData = se1@colData,
                                     design = ~ characteristics_ch1.3,
                                     metadata = metadata(se1))
dds_sex <- DESeqDataSetFromMatrix(countData = df1b,
                                     colData = se1@colData,
                                     design = ~ characteristics_ch1.5,
                                     metadata = metadata(se1))
dds_sex_p_diet <- DESeqDataSetFromMatrix(countData = df1b,
                                     colData = se1@colData,
                                     design = ~ characteristics_ch1.5 + characteristics_ch1.3,
                                     metadata = metadata(se1))
```
### Saving to Rdata file
```{r}
saveRDS(
  list("diet" = dds_diet,
       "sex_p_diet" = dds_sex_p_diet,
       "sex" = dds_sex),
  file = "data.rds")

ddslist <- readRDS(file = "data.rds")
```

## Exploratory analysis and visualisation

### Transformation

```{r}
dds_diet_filtered <- dds_diet[rowSums(counts(dds_diet)) > 1,]

dds_diet_vst <- vst(dds_diet_filtered, blind = F)
dds_diet_rld <- rlog(dds_diet_filtered, blind = F)
```
#### Effect

```{r}
dds_diet_est_factors <- estimateSizeFactors(dds_diet)

as.data.frame(
  log2(
    counts(dds_diet_est_factors,
           normalized = TRUE)[,1:2]+1)) %>% 
  mutate(transformation = "log2(x + 1)") %>% 
  bind_rows(as_data_frame(assay(dds_diet_vst)[, 1:2]) %>% mutate(transformation = "vst"),
            as_data_frame(assay(dds_diet_rld)[, 1:2]) %>% mutate(transformation = "rlog")) -> dds_diet_dataframe

colnames(dds_diet_dataframe)[1:2] <- c("x","y")

dds_diet_dataframe_levels <- c("log2(x + 1)", "vst", "rlog")
dds_diet_dataframe$transformation <- factor(dds_diet_dataframe$transformation, levels = dds_diet_dataframe_levels)

```

#### Plot

```{r}
ggplot(dds_diet_dataframe,
       aes(x, y)) +
  geom_hex(bins = 80) +
  coord_fixed() +
  facet_grid(. ~ transformation)
```

#### Sample Distances

```{r}
dds_diet_dist <- dist(t(assay(dds_diet_vst)))

dds_diet_dist_matrix <- as.matrix(dds_diet_dist)
rownames(dds_diet_dist_matrix) <- paste(dds_diet_vst$characteristics_ch1.3,
                                        sep = " - ")
colnames(dds_diet_dist_matrix) <- NULL

colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)

pheatmap(dds_diet_dist_matrix,
         clustering_distance_rows = dds_diet_dist,
         clustering_distance_cols = dds_diet_dist,
         col = colors)
```

#### PCA plot

```{r}
plotPCA(dds_diet_vst, intgroup = c("characteristics_ch1.3","characteristics_ch1.5"))
```

#### MDS

```{r}
dds_diet_mds <- as.data.frame(colData(dds_diet_vst)) %>% 
  cbind(cmdscale(dds_diet_dist_matrix))
```

```{r}
ggplot(dds_diet_mds, 
       aes(x = `1`,
           y = `2`,
           color = characteristics_ch1.3,
           shape = characteristics_ch1.5)) +
  geom_point(size = 3) + 
  coord_fixed() +
  ggtitle("MDS with VST data")
```

```{r}
ggplot(dds_diet_mds, 
       aes(x = `1`,
           y = `2`,
           color = title)) +
  geom_point(size = 3) + 
  coord_fixed() +
  ggtitle("MDS with VST data")
```

### Differential Expression Analysis

```{r}
DESeq_diet <- DESeq(dds_diet)

DESeq_diet_res <- results(DESeq_diet)
DESeq_diet_res_compare <- results(DESeq_diet,
                             contrast = c("characteristics_ch1.3","chow",""))

```


***
# TODO
- <input type="checkbox" checked>todo</input>

- <input type="checkbox" >Brief description of article / aims of study</input>

- <input type="checkbox"> interpretation of the visualisations (heatmap, MDS, etc.)</input>

- <input type="checkbox"> Differential expression analysis</input>

- <input type="checkbox"> presentation of the results of the functional enrichment analysis </input>

  - (identify some biological processes from GO BP or KEGG pathways identified by ORA or GSEA analysis that are important for the organism response/disease you are investigating).
  
- <input type="checkbox"> compare your results with the results reported in the article and highlight your main conclusions</input>

- bonus
  - genes with the highest value
  - genes with the lowest value
  - include info on gender (only for visualization)
```{r}
se1@colData$characteristics_ch1.5

```

```{r}
se1@colData$characteristics_ch1.3
```




# TL;DR

Wenge Li and colleagues, the findings include:

- Diet choice can significantly affect stem cell function, including proliferation, differentiation, and senescence.

- Different types of diets, such as high-fat diets, low-calorie diets, and fasting, have different effects on stem cells.

- The effects of diet on stem cells are mediated by various mechanisms, such as oxidative stress, inflammation, and epigenetic changes.

- The choice of animal models and cell types used in stem cell research can also affect the interpretation of the results and the generalizability of the findings.

- The quality and clarity of reporting in scientific publications are crucial for facilitating reproducibility and transparency in stem cell research, especially regarding the details of diet selection and administration.
